mopidy/audio/actor.py

```diff
@@ -735,35 +732,44 @@ def set_metadata(self, track):
         """
         Set track metadata for currently playing song.
 
-        Only needs to be called by sources such as `appsrc` which do not
+        Only needs to be called by sources such as ``appsrc`` which do not
         already inject tags in playbin, e.g. when using :meth:`emit_data` to
         deliver raw audio data to GStreamer.
 
         :param track: the current track
         :type track: :class:`mopidy.models.Track`
         """
-        taglist = gst.TagList()
+        taglist = Gst.TagList.new_empty()
         artists = [a for a in (track.artists or []) if a.name]
 
+        def set_value(tag, value):
+            gobject_value = GObject.Value()
+            gobject_value.init(GObject.TYPE_STRING)
+            gobject_value.set_string(value)
+            taglist.add_value(
+                Gst.TagMergeMode.REPLACE, Gst.TAG_ARTIST, gobject_value)
+
         # Default to blank data to trick shoutcast into clearing any previous
```

Comment:
To be honest I think this doesn't actually work. I've tried playing with this using my hand made playbin alternative test code and this did not seem effective. For this PR I would just drop a TODO to double check.